app/components/modelling/galaxyModel.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable class-methods-use-this */
+import baseModel from './baseModel';
+import galaxyRegls from './galaxyRegls';
+import postProcessingRegl from './postProcessingRegl';
+
+class galaxyModel extends baseModel {
+  getModel() {
+    return [
+      {
+        name: 'sersic disk',
+        func: galaxyRegls.drawSersic,
+        map: ['mux', 'muy', 'rx', 'ry', 'roll', 'scale', 'i0'],
+        type: 'component',
+        default: { mux: 0, muy: 0, rx: 10, ry: 15, scale: 5 / 8, roll: 0, i0: 0.75, n: 1, c: 2 }
+      },
+      {
+        name: 'sersic bulge',
+        func: galaxyRegls.drawSersic,
+        map: ['mux', 'muy', 'rx', 'ry', 'roll', 'scale', 'i0', 'n'],
+        type: 'component',
+        default: { mux: 100, muy: 100, rx: 10, ry: 15, scale: 5 / 8, roll: 0, i0: 0.75, n: 1, c: 2 }
+      },
+      {
+        name: 'sersic bar',
+        func: galaxyRegls.drawSersic,
+        map: ['mux', 'muy', 'rx', 'ry', 'roll', 'scale', 'i0', 'n', 'c'],
+        type: 'component',
+        default: { mux: 100, muy: 100, rx: 5, ry: 5, scale: 5 / 8, roll: 0, i0: 0.75, n: 2, c: 2 }
+      },
+      {
+        name: 'sersic spiral',
+        func: galaxyRegls.drawSpiral,
+        map: ['points', 'spread', 'i0'],
+        type: 'component',
+        default: { mux: 100, muy: 100, rx: 5, ry: 5, spread: 1, roll: 0, i0: 0.75, n: 2, c: 2, falloff: 1 }
+      }
+    ];
+  }
+  parse(annotation) {
+    // TODO: make this explicitly related to the galaxy rendering model.
+    // TODO: link spiral arm to disk? Link centres?
+    // we have disk, then bulge then bar
+    const renderFunctions = [];
+    const annotationLength = annotation.length;
+    let disk = null;
+    if (annotationLength > 0 && annotation[0].value[0].value.length > 0) {
```

Comment:
Having the `annotation` index for each of the model components does not seem like a good idea.  If the order of the workflow questions change you should not have to change the drawing code.

We should have a way to auto-detect this from the workflow structure.